provisioner/converge/provisioner.go

```diff
@@ -0,0 +1,262 @@
+
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	"github.com/mitchellh/packer/template/interpolate"
+)
+
+var versionRegex = regexp.MustCompile(`^[\.\-\da-zA-Z]*$`)
+
+// Config for Converge provisioner
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+
+	// Bootstrapping
+	Bootstrap bool   `mapstructure:"bootstrap"`
+	Version   string `mapstructure:"version"`
+
+	// Modules
+	ModuleDirs []ModuleDir `mapstructure:"module_dirs"`
+	Modules    []Module    `mapstructure:"modules"`
+
+	ctx interpolate.Context
+}
+
+// ModuleDir is a directory to transfer to the remote system
+type ModuleDir struct {
+	Source      string   `mapstructure:"source"`
+	Destination string   `mapstructure:"destination"`
+	Exclude     []string `mapstructure:"exclude"`
+}
+
+// Module contains information needed to run a module
+type Module struct {
+	Module    string            `mapstructure:"module"`
+	Directory string            `mapstructure:"directory"`
+	Params    map[string]string `mapstucture:"params"`
+}
+
+// Provisioner for Converge
+type Provisioner struct {
+	config Config
+}
+
+// Prepare provisioner somehow. TODO: actual docs
+func (p *Provisioner) Prepare(raws ...interface{}) error {
+	err := config.Decode(
+		&p.config,
+		&config.DecodeOpts{
+			Interpolate:        true,
+			InterpolateContext: &p.config.ctx,
+		},
+		raws...,
+	)
+	if err != nil {
+		return err
+	}
+
+	// validate version
+	if !versionRegex.Match([]byte(p.config.Version)) {
+		return fmt.Errorf("Invalid Converge version %q specified. Valid versions include only letters, numbers, dots, and dashes", p.config.Version)
+	}
+
+	// validate sources and destinations
+	for i, dir := range p.config.ModuleDirs {
+		if dir.Source == "" {
+			return fmt.Errorf("Source (\"source\" key) is required in Converge module dir #%d", i)
+		}
+		if dir.Destination == "" {
+			return fmt.Errorf("Destination (\"destination\" key) is required in Converge module dir #%d", i)
+		}
+	}
+
+	// validate modules
+	if len(p.config.Modules) == 0 {
+		return errors.New("Converge requires at least one module (\"modules\" key) to provision the system")
+	}
+	for i, module := range p.config.Modules {
+		if module.Module == "" {
+			return fmt.Errorf("Module (\"module\" key) is required in Converge module #%d", i)
+		}
+		if module.Directory == "" {
+			p.config.Modules[i].Directory = "/tmp"
+		}
+	}
+
+	return err
+}
+
+// Provision node somehow. TODO: actual docs
+func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
+	ui.Say("Provisioning with Converge")
+
+	// bootstrapping
+	if err := p.maybeBootstrap(ui, comm); err != nil {
+		return err // error messages are already user-friendly
+	}
+
+	// check version (really, this make sure that Converge is installed before we try to run it)
+	if err := p.checkVersion(ui, comm); err != nil {
+		return err // error messages are already user-friendly
+	}
```

Comment:
It does fail if it's not on the path, do you mean not checking the version at all?